### PR TITLE
Make vesting contract creation parsing more resilient

### LINKED ISSUE #2690: Related addresses on vesting contract creation are wrong
https://github.com/nimiq/core-rs-albatross/blob/5cfa101819663f7ff4a2151346dfc75ea83fe2e2/primitives/transaction/src/lib.rs#L477-L478

`Deserialize::deserialize_from_vec` of this struct is wrong and can't be used here.

### PR DESCRIPTION
**Pull request by hrxi**
Use `serde` to do it, but remove the `Deserialize` impl because it was error-prone.

Based on top of #2676, so this should wait until #2676 is approved.

Fixes #2690.

### CODE REVIEW

File: primitives/transaction/src/account/vesting_contract.rs [hunk 1]
```diff
@@ -68,79 +68,132 @@ impl AccountTransactionVerification for VestingContractVerifier {
     }
 }
 
-#[derive(Default, Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Default)]
 pub struct CreationTransactionData {
```
Comment (jsdanielh) [inline primitives/transaction/src/account/vesting_contract.rs:65]: Could you add some rustdoc?

File: primitives/transaction/src/account/vesting_contract.rs [hunk 2]
```diff
@@ -68,79 +68,132 @@ impl AccountTransactionVerification for VestingContractVerifier {
...
-        if transaction.recipient_data.len() == Address::SIZE + 8 {
+    fn parse_impl(data: &[u8], tx_value: Coin) -> Result<Self, TransactionError> {
+        // TODO: Change this into `const` block once Rust 1.79 is sufficiently old.
+        #[allow(clippy::assertions_on_constants)]
+        {
+            assert!(CreationTransactionData8::SIZE != CreationTransactionData24::SIZE);
+            assert!(CreationTransactionData8::SIZE != CreationTransactionData32::SIZE);
+            assert!(CreationTransactionData24::SIZE != CreationTransactionData32::SIZE);
+        }
+        Ok(if data.len() == CreationTransactionData8::SIZE {
```
Comment (jsdanielh) [inline primitives/transaction/src/account/vesting_contract.rs]: Why are we not using here a `match` statement?

File: serde/src/lib.rs [hunk 3]
```diff
@@ -91,14 +91,22 @@ pub trait SerializedMaxSize {
     const MAX_SIZE: usize;
 }
 
+/// Types implementing this trait have an alternative fixed-size serialization length.
+pub trait SerializedFixedSize {
```
Comment (jsdanielh) [inline serde/src/lib.rs:95]: I think a better naming could be `FixedSerializedSize`. wdyt?

File: primitives/transaction/src/account/vesting_contract.rs (hunk 2 above)
Comment (hrxi) [reply primitives/transaction/src/account/vesting_contract.rs]: Fixed.

File: primitives/transaction/src/account/vesting_contract.rs (hunk 1 above)
Comment (hrxi) [reply primitives/transaction/src/account/vesting_contract.rs:65]: Done.

File: serde/src/lib.rs (hunk 3 above)
Comment (hrxi) [reply serde/src/lib.rs:95]: I think I prefer the current name.

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**